framework/doc/content/syntax/Postprocessors/index.md

```diff
@@ -1,10 +1,170 @@
+
+This first step of coding this type of coupling begins by adding the necessary input file syntax to
+the object that requires a postprocessor value, PostprocessorNeumannBC in this example. In all MOOSE
+objects input file syntax is governed by the validParams function of an object. To add the ability
+to couple a postprocessor, simply add a new parameter using the `PostprocessorName` type, as shown
+below. Notice, that the add parameters call includes a default value that makes the use of the
+postprocessor optional.
+
+!listing PostprocessorNeumannBC.C start=template end=PostprocessorNeumannBC::
+
+The actual postprocessor value must be assigned to a member variable of the class, thus in the header
+a member variable must be created, which should always be a constant reference to a
+`PostprocessorValue` type. Since this is a reference it must be initialized, this occurs in the
+source file by calling the `getPostprocessorValue` method and providing the name used in the
+validParams function. The following snippets show declaration of the reference in the header and
+the initialization of this reference in the source file.  The `_value` member variable is then
+available for use anywhere inside the object, for the case of the boundary condition it is utilized
+in the computation of the residual.
+
+!listing PostprocessorNeumannBC.h line=PostprocessorValue
+
+!listing PostprocessorNeumannBC.C start=PostprocessorNeumannBC:: end=} include-end=true
+
+## Creating a `Postprocessor` Object
+
+In general, every Postprocessor object has two methods that must be defined "execute" and
+"getValue".
+
+First, consider the execute method. This method is called by MOOSE at different time
+depending on the type of postprocessor object. Therefore, when creating a Postprocessor object
+the new object should inherit from one of the following C++ classes:
+
+- +GeneralPostprocessor+: "execute" is called once on each execution flag.
+- +NodalPostprocessor+: "execute" is called for each +node+ within the mesh on each execution flag.
+- +ElementalPostprocessor+: "execute" is called for each +element+ within the mesh on each execution
+   flag.
+- +InternalSidePostprocessor+: "execute" is called for each +side+, that is not on a boundary,
+   within the mesh on each execution flag.
+- +SidePostprocessor+: "execute" is called for each +side+, that is on a boundary, within the mesh
+   on each execution flag.
+
+The use of execution flags is discussed in the [Execute On](#execute-on) section.
+
+The getValue method is responsible for returning the value of the postprocessor object, this
+value is what is used by all objects that are coupled to the postprocessor.
+
+### Parallel Considerations
+
+When creating a postprocessor it is often necessary to perform some parallel communication
+to ensure that the value being computed is consistent across processes and threads. Three additional
+methods exists for making this process as simple as possible.
```

Comment:
I like this but "consistent" is a little vague. I usually like to talk about "aggregating" the value since that's usually what we are doing. You can leave it as is, or change it. This is a good start, regardless.

Reply:
How about "correct", I use the word aggregate to describe a PP in general and don't want this to be confused with the PP definition.